src/Jira/Api.php

```diff
@@ -875,14 +882,23 @@ public function getProjectIssueTypes($project_key)
 	/**
 	 * Returns a list of all resolutions.
 	 *
-	 * @param string $project_key Project key.
-	 *
-	 * @return array|false
+	 * @return array
 	 * @since  2.0.0
 	 */
 	public function getResolutions()
 	{
-		return $this->api(self::REQUEST_GET, '/rest/api/2/resolution', array(), true);
+		if ( !count($this->resolutions) ) {
```

Comment:
Considering that default value of `resolutions` property is `null` instead of empty array it's not good to treat it as array and try to count values in it. Maybe it even results in a notice.

I think, that assigning an empty array to `resolutions` property would be correct solution here.

P.S.

If you wish you can do the same with `fields`, `priorities` and `statuses` properties, that are used the same way.